auth-service/src/main/java/cab/aggregator/app/authservice/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,210 @@
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static cab.aggregator.app.authservice.util.Constants.AUTH_TOKEN;
+import static cab.aggregator.app.authservice.util.Constants.CLIENT_ID_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.CLIENT_SECRET_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.DRIVER_ROLE;
+import static cab.aggregator.app.authservice.util.Constants.GENDER_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.GRANT_TYPE_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.GRANT_TYPE_PASSWORD_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.PASSENGER_ROLE;
+import static cab.aggregator.app.authservice.util.Constants.PASSWORD_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.REFRESH_TOKEN_FIELD;
+import static cab.aggregator.app.authservice.util.Constants.SERVICE_UNAVAILABLE_MESSAGE;
+import static cab.aggregator.app.authservice.util.Constants.USERNAME_FIELD;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final Keycloak keycloak;
+    private final RestTemplate restTemplate;
+    private final KeycloakProperties keycloakProperties;
+    private final MessageSource messageSource;
+    private final ObjectMapper objectMapper;
+    private final PassengerClientContainer passengerClientContainer;
+    private final DriverClientContainer driverClientContainer;
+
+    @Override
+    public void signUp(SignUpDto signUpDto) {
+        UserRepresentation keycloakUser = getUserRepresentation(signUpDto);
+        RealmResource realmResource = keycloak.realm(keycloakProperties.getRealm());
+        UsersResource usersResource = realmResource.users();
+        String adminClientAccessToken = keycloak.tokenManager().getAccessTokenString();
+        Response response;
+        response = Optional.ofNullable(usersResource.create(keycloakUser))
+                .orElseThrow(() -> new ServiceUnavailableException(messageSource.getMessage(SERVICE_UNAVAILABLE_MESSAGE, new Object[]{}, LocaleContextHolder.getLocale())));
```

Comment:
Зачем было разделять объвление и инициализацию переменной?
```suggestion
        Response response = Optional.ofNullable(usersResource.create(keycloakUser))
                .orElseThrow(() -> new ServiceUnavailableException(messageSource.getMessage(SERVICE_UNAVAILABLE_MESSAGE, new Object[]{}, LocaleContextHolder.getLocale())));
```